src/utils/fetchClient.ts

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+const BASE_URL = 'https://mate.academy/students-api';
+
+// returns a promise resolved after a given delay
```

Comment:
Try to avoid such commenting of the code, it is better to name the functions so that they explain what they do through their name, for example: `waitPromiseResolving` or `waitRequestResolving`.